src/main/kotlin/parts/wisdom/arcraider/Coord.kt

```diff
@@ -0,0 +1,47 @@
+package parts.wisdom.arcraider
+
+import me.joypri.Mix
+import me.joypri.to
+import me.joypri.Part
+import me.joypri.Role
+
+object X : Role<Int>()
+object Y : Role<Int>()
+object Width: Role<Int>()
+object Height: Role<Int>()
+object Coords: Role<List<Coord>>()
+object TheGrid: Role<Grid>()
+
+open class Coord(vararg parts: Part) : Mix(*parts) {
+    val x by X
+    val y by Y
+}
+
+open class Grid(vararg parts:Part) : Mix(*parts) {
```

Comment:
Is this intended to be the main Grid class for problem solving? (I looked at the package and file for the answer, but it's in the top-level arcraider package and why is it in Coord.kt?) To me, this seems like an odd representation for the main Grid class:
* Although I strongly lean toward immutable data structures, I wonder if Grid needs to be mutable. We'll end up writing lots of imps that essentially render into Grids.
* To me, Grid has the flavor of a regular Kotlin class instead of a Mix. For example, I'm not expecting to need to construct similar data structures dynamically, to sometimes have only a subset of these fields (e.g. to have a Width but not a Height), to sometimes have additional fields (a Depth?), etc.
* I'm thinking of Grid as a fairly performance-critical class. We'll be doing things like searching through alternative clans that might generate a given Grid and feeding Grids into neural nets.
* I'm thinking of the most common Grid operations as "set (x, y) to c" and "get (x, y)", which aren't natural to implement on List<Coord>.

Reply:
Seperated Grid concepts